preprocess_era5_modellevels.py

```diff
@@ -0,0 +1,178 @@
+    lower_layer = dp_modellevels.lev > boundary 
+    upper_layer = ~lower_layer
+
+    # Integrate fluxes and state
+    fa_e_lower = fa_e.where(lower_layer).sum(dim="lev")
+    fa_n_lower = fa_n.where(lower_layer).sum(dim="lev")
+    w_lower = cwv.where(lower_layer).sum(dim="lev")
+
+    fa_e_upper = fa_e.where(upper_layer).sum(dim="lev")
+    fa_n_upper = fa_n.where(upper_layer).sum(dim="lev")
+    w_upper = cwv.where(upper_layer).sum(dim="lev")
+   
+    print(
+        "Check calculation water vapor over two layers, this value should be zero:",
+        (cwv.sum(dim="lev") - (w_upper + w_lower)).sum().values,
+    )
+
+    tcwm3 = tcw* a_gridcell[np.newaxis,:] / density_water # m3
+    
+    print(
+        "Check calculation water vapor with column water vapour, this value should be close to zero:",
+        (cwv.sum(dim="lev") - (tcwm3)).sum().values,
+    )
+    # is niet hetzelfde.. hoe erg is dat?
+    
+    # Change units to m3
+    # TODO: Check this! Change units before interp is tricky, if not wrong
+    total_seconds = config["timestep"] / config["divt"]
+    fa_e_upper *= total_seconds * (l_ew_gridcell / density_water)
+    fa_e_lower *= total_seconds * (l_ew_gridcell / density_water)
+    fa_n_upper *= total_seconds * (l_mid_gridcell[None, :, None] / density_water)
+    fa_n_lower *= total_seconds * (l_mid_gridcell[None, :, None] / density_water)
+
+    # Put data on a smaller time step...
+    time = w_upper.time.values
+    newtime = pd.date_range(time[0], time[-1], freq="15Min")[:-1]
+    w_upper = w_upper.interp(time=newtime).values
+    w_lower = w_lower.interp(time=newtime).values
+
+    # ... fluxes on the edges instead of midpoints
+    newtime = newtime[:-1] + pd.Timedelta("6Min") / 2
+    fa_e_upper = fa_e_upper.interp(time=newtime).values
+    fa_n_upper = fa_n_upper.interp(time=newtime).values
+
+    fa_e_lower = fa_e_lower.interp(time=newtime).values
+    fa_n_lower = fa_n_lower.interp(time=newtime).values
+
+    precip = (precip.reindex(time=newtime, method="bfill") / 4).values
+    evap = (evap.reindex(time=newtime, method="bfill") / 4).values
+
+    # Stabilize horizontal fluxes
+    fa_e_upper, fa_n_upper = get_stable_fluxes(fa_e_upper, fa_n_upper, w_upper)
+    fa_e_lower, fa_n_lower = get_stable_fluxes(fa_e_lower, fa_n_lower, w_lower)
+
+    # Determine the vertical moisture flux
+    fa_vert = get_vertical_transport(
+        fa_e_upper,
+        fa_e_lower,
+        fa_n_upper,
+        fa_n_lower,
+        evap,
+        precip,
+        w_upper,
+        w_lower,
+    )
+
+    # Save preprocessed data
+    # Note: fluxes (dim: time) are at the edges of the timesteps,
+    # while states (dim: time2) are at the midpoints and include next midnight
+    # so the first state from day 2 will overlap with the last flux from day 1
+    filename = f"{date.strftime('%Y-%m-%d')}_fluxes_storages.nc"
+    output_path = os.path.join(config["interdata_folder"], filename)
```

Comment:
interdata_folder has been renamed to preprocessed_data_folder

output_path = os.path.join(config["preprocessed_data_folder"], filename)